src/components/carousel/index.js

```diff
@@ -145,11 +145,31 @@ export default class Carousel extends BaseComponent {
     return index;
   }
 
+  getSnapToOffsets = () => {
+    const {itemSpacings} = this.props;
+    const {containerWidth, pageWidth} = this.state;
+
+    if (containerWidth) {
```

Comment:
This condition means that even if the user passed 'pageWidth' it needs to wait for the onLayout to get the snapToOffset.

Reply:
it will have to wait for the `snapToOffset` because the code of the offsets depends on the containerWidth. 
That's another issue we had there, we always assume the carousel is full-width, so the "breakpoints" were in-correct sometimes. 
But it doesn't delay the render. I checked it.

Reply:
Yes sure, you can clearly see that it doesn't delay the render, especially with the other carousel example on the screen.